Modules/ArcGIS/ArcGIS.psm1

```diff
@@ -259,53 +247,37 @@ function Get-ArcGISURL
         $Role = $AllNodes[$i].Role
         if($Role -icontains 'Server' -and -not($PrimaryServerMachine))
         {
-            $PrimaryServerMachine  = $AllNodes[$i]
-            $PrimaryServerMachineName  = $PrimaryServerMachine.NodeName
+            $PrimaryServerMachineNode  = $AllNodes[$i]
         }
 
         if($Role -icontains 'Portal' -and -not($PrimaryPortalMachine))
         {
-            $PrimaryPortalMachine= $AllNodes[$i]
-            $PrimaryPortalMachineName = $PrimaryPortalMachine.NodeName
+            $PrimaryPortalMachineNode= $AllNodes[$i]
         }
     }
 
-    if($PrimaryPortalMachine)
+    if($PrimaryPortalMachineNode)
     {
-        $PortalExternalDNSName = Get-FQDN $PrimaryPortalMachineName
+        $PortalExternalDNSName = Get-FQDN $PrimaryServerMachineNode.NodeName
```

Comment:
Shouldn't this be
` $PortalExternalDNSName = Get-FQDN $PrimaryPortalMachineNode.NodeName `
?

Reply:
Yep, Will fix this.